chain/network/src/peer/peer_actor.rs

```diff
@@ -556,14 +558,15 @@ impl PeerActor {
         // decides whether to accept the connection or not: ctx.wait makes
         // the actor event loop poll on the future until it completes before
         // processing any other events.
-        ctx.wait(wrap_future(self.network_state.peer_manager_addr
-                .send(PeerToManagerMsg::RegisterPeer(RegisterPeer {
-                    connection: conn.clone(),
-                }).with_span_context())
-            )
-            .then(move |res, act: &mut PeerActor, ctx| {
-                match res.map(|r|r.unwrap_consolidate_response()) {
-                    Ok(RegisterPeerResponse::Accept) => {
+        ctx.wait(wrap_future({
+            let network_state = self.network_state.clone();
+            let clock = self.clock.clone();
+            let conn = conn.clone();
+            async move { network_state.register(&clock,conn).await }
```

Comment:
Hm, isn't this just ` network_state.register(&clock, conn)` ?

Reply:
it would be iff register() was a method a method of Arc<Self>, rather than of &Arc<Self>. Previously when we talked about it, your opinion was that stuff like network_state.clone().register(...) looks weird, so I was sticking to &Arc<Self> since then.